lib/ProductOpener/APIProductWrite.pm

```diff
@@ -85,6 +86,20 @@ sub skip_protected_field ($product_ref, $field, $moderator = 0) {
 	return 0;
 }
 
+=head2 skip_protected_image($product_ref, $imagefield, $moderator = 0)
+
+=cut
+
+sub is_protected_image ($product_ref, $imagefield, $moderator = 0) {
```

Comment:
I think we should remove the $moderator from the parameter: the function should indicate if the image is protected or not. Then if an user is a moderator, we can decide not to check if the image is protected.